dspace-api/src/main/java/org/dspace/authenticate/ShibAuthentication.java

```diff
@@ -332,6 +328,8 @@ public int authenticate(Context context, String username, String password,
 							affiliation = affiliation.substring(index+1, affiliation.length());
 						}
 					} 
+                                        // Replace spaces with underscores
+					affiliation = affiliation.replace(" ", "_");
```

Comment:
This seems like a change in behavior that should only occur in a major release. Currently, I believe you can specify group names with spaces by escaping the space in your config file (e.g. "My\ Group\ with\ spaces").  This behavior change would mean that such existing configurations would need updated (so it'd need documenting as part of an upgrade process).